docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,271 @@
+InternalSubscribeResponseTopic = “edgex/command/response/#”
+ExternalPublishResponseTopicPrefix = “edgex/command/response/"  # /<device-name>/<command-name>/<method> will be added to this publish topic prefix
+```
+
+#### device service configuration
+
+Example device service configuration is provided below.
+
+``` toml
+[MessageQueue]
+Protocol = "redis"
+Host = "localhost"
+Port = 6379
+Type = "redis"
+AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+SecretName = "redisdb"
+SubscribeRequestTopic = "edgex/command/request/#"
+PublishResponseTopicPrefix = “edgex/command/response/"  # <device-service>/<device-name>/<command-name>/<method> will be added to this publish topic prefix
+```
+
+## Questions
+
+- Do we need separate topics for all the devices or would one on the device service suffice?
+    
+    - Ans:  we have defined the deviceName in the payload, so one topic should be sufficient for Device Service- edgex/command/request/<device-name>...
+
+- Would clients (non EdgeX services and applications) want to get a list of available commands via message (instead of calling REST)?
+
+    - Ans:  this is a valid question and could be provided via later additions to the command service (or other service like metadata) in the future.  It does not have to be tackled immediately.  
+
+- Dynamic configuration of the message subscription is not a user friendly operation today (requiring configuration changes).
+
+    - Ans:  We might want to think about creating additional APIs for Adding/Updating/Deleting/Query the external subscription (and store them to the RedisDB).
+
+- Is it acceptable for more than one response to be published by the device service on the same correlation ID? Eg, send back "Acknowledged", then "Scheduled", then "Starting", then "Done" statuses?
+    
+    - Ans:  No, the correlation id has a life span to/from the initial requester to the response back to the requester.
+
+- Would it make sense to echo the command name into the response, as a reality check?
+    - Ans: solved via topic naming
+
+- Would sending/receiving binary data (e.g. CBOR) be supported in this north-south message implementation?
+    
+    - Ans:  today, command service and device services do not support CBOR operations.  This feature would be explored only when these services support binary/CBOR payloads.
+
+- Use of the message bus communications (by the non-EdgeX 3rd party service or application) would bypass the API Gateway.
+
+    - Ans:  it would. Pretty much every MQTT broker has a way to do this, where the most common are username/password and certificate-based mutual-auth TLS. SPIFFE/SPIRE is a good way to distribute such certificates if one can manage the requirements for certificate rotation on a continual basis. Otherwise, username/password will probably end up being used (with server-side TLS if the broker is off-box)
```

Comment:
No longer an issue with the change to Command Service having a External MQTT bridge connection.